src/components/calcite-action/calcite-action.scss

```diff
@@ -1,246 +1,230 @@
 }
 
 :host([disabled]) {
-  pointer-events: none;
+  @apply pointer-events-none;
 }
 
 .button {
+  @apply bg-foreground-1
+    border-none
+    cursor-pointer
+    fill-color-3
+    flex
+    focus-base
+    items-center
+    justify-start
+    m-0
+    relative
+    text--2h
+    text-color-3;
   font-family: inherit;
-  display: flex;
-  color: var(--calcite-ui-text-3);
-  fill: var(--calcite-ui-text-3);
-  background-color: var(--calcite-ui-foreground-1);
-  margin: 0;
-  justify-content: flex-start;
-  align-items: center;
-  border: none;
   width: auto;
-  cursor: pointer;
   text-align: unset;
-  position: relative;
-  font-size: theme("fontSize.-2");
-  @include focus-style-base();
-  &:hover,
-  &:focus {
-    background-color: var(--calcite-ui-foreground-2);
-    color: var(--calcite-ui-text-1);
-    fill: var(--calcite-ui-text-1);
+  &:hover {
+    @apply bg-foreground-2 text-color-1 fill-color-1;
   }
 
   &:focus {
-    background-color: var(--calcite-ui-foreground-2);
-    color: var(--calcite-ui-text-1);
-    fill: var(--calcite-ui-text-1);
-    @include focus-style-inset();
+    @apply bg-foreground-2 text-color-1 fill-color-1 focus-inset;
   }
 
   &:active {
-    background-color: var(--calcite-ui-foreground-3);
+    @apply bg-foreground-3;
   }
 
   .icon-container {
-    min-width: 1rem;
-    min-height: 1rem;
-    margin: 0;
-    justify-content: center;
-    align-items: center;
-    display: flex;
-    pointer-events: none;
+    @apply flex
+      items-center
+      justify-center
+      m-0
+      pointer-events-none;
+    min-width: theme("spacing.4");
+    min-height: theme("spacing.4");
   }
 
   .text-container {
-    color: inherit;
-    line-height: 1rem;
-    white-space: nowrap;
-    overflow: hidden;
-    text-overflow: ellipsis;
-    margin: 0;
-    opacity: 0;
-    width: 0;
-    transition: opacity $transition, margin $transition, width $transition;
+    @apply leading-none
+      truncate
+      m-0
+      w-0
+      opacity-0
+      ease-in-out
+      duration-150
+      transition-opacity;
+    transition-property: margin;
+    transition-property: width;
   }
 
   .text-container--visible {
-    flex: 1 1 auto;
-    opacity: 1;
+    @apply flex-auto opacity-100;
     width: auto;
   }
 }
 
 :host([scale="s"]) .button {
-  padding: var(--calcite-spacing-half) var(--calcite-spacing-half);
+  @apply p-2;
 }
 
 :host([scale="m"]) .button {
-  padding: var(--calcite-spacing) var(--calcite-spacing);
+  @apply p-4;
 }
 
 :host([scale="l"]) .button {
-  padding: var(--calcite-spacing-plus-quarter) var(--calcite-spacing-plus-quarter);
-  @include font-size(-1);
+  @apply p-5 text--1h;
 }
 
 // Compact
 :host([scale="s"][compact]) .button,
 :host([scale="m"][compact]) .button,
 :host([scale="l"][compact]) .button {
-  padding-left: 0;
-  padding-right: 0;
+  @apply px-0;
 }
 
 .slot-container {
-  display: flex;
+  @apply flex;
 }
 
 .slot-container--hidden {
-  display: none;
+  @apply hidden;
```

Comment:
What would be the tailwind rule for `visibility: hidden` if `hidden` corresponds to `display: none`?  Asking for a friend... ;)

Reply:
`invisible` :exploding_head: